### PR TITLE
refactoring further steps

### PR DESCRIPTION
**Pull request by JulianGartner**
removed some code, moved some files,

### CODE REVIEW

**Comment by cchet** [comment]
@JulianGartner @rainer-gepardec 

This pull request will be merged when the other pull request have been merged, so prepare for a merge party :)

**Comment by JulianGartner** [comment]
@cchet I planned on merging this one last

Reviews nevertheless can be done in the meantime ;)

**Comment by cchet** [comment]
@rainer-gepardec @JulianGartner 

**Please review my changes regarding mailing:**
1. Extracted subjects to message.properties which we access via a ResourceBundle instance
2. Message keys contain Reminder enumeration name and are dynamically resolved
3. Moved notification configurations to NotificationConfig class which were in MailSender class
4. Fixed affected tests. Tests no hard code the expected subject
5. Deleted subject config parameters from application.properties
6. Template paths in application.properties contain Reminder enumeration name as well and are also dynamically resolved

**Affected files:**

MailSender, NotificationConfig, MailSenderTest, MailDaemonTest, ResourceBundleProducer
src/main/resource/message.properties
src/main/resource/application.properties

**Comment by cchet** [comment]
@rfischerGep   @rainer-gepardec 

When build is done I would like to merge. Do you all approve this PR

File: mega-zep-backend/src/main/java/com/gepardec/mega/application/messages/ResourceBundleProducer.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import java.util.Locale;
+import java.util.ResourceBundle;
+
+@ApplicationScoped
+public class ResourceBundleProducer {
+
+    private ResourceBundle mailMessages;
```
Comment (rfischerGep) [inline mega-zep-backend/src/main/java/com/gepardec/mega/application/messages/ResourceBundleProducer.java:11]: field is unused?

Comment (cchet) [reply mega-zep-backend/src/main/java/com/gepardec/mega/application/messages/ResourceBundleProducer.java]: fixed and commited

**Pull request merged**